doc/api/esm.md

```diff
@@ -423,9 +423,6 @@ thrown:
 
 ### Dual CommonJS/ES Module Packages
```

Comment:
This is the section we want to point people to, when answering the question "how to write packages that are both CJS and ESM?". The first code example should thus be something copyable.

Reply:
Well maybe we should do without a code example then for the hazard explanation. I can just explain the hazard abstractly without any “bad” code for people to copy. The hazard needs to be introduced before the recommended approaches so that the explanations for why they're recommended make sense.

Reply:
That sounds sensible to me.